Better Keyboard A11y + Navigation

aryanpingle opened the pull request:
Keyboard accessibility is super important, and not just for the visually impaired. It's common practice to `TAB` through a website when your mouse is faulty and/or you can't be bothered to reach for it. The options components  aren't very keyboard-accessible; this PR aims to change that.

Changes include:
* Making all input types react to keyboard focus
* Minor design improvements in this scope

## Visual Comparison

In both before and after I `TAB` to the `Advanced Settings` expander, and then `TAB` to the end. If it feels like I'm not doing anything in the _before changes_ video... well, that's the issue. I also try to make the checkboxes feel like Material UI, but I can only get so far without a complete overhaul.

[Before changes - Little to no focus effects](https://imgur.com/a/eXLFNsR)
[After changes - 100% support for keyboard navigation](https://imgur.com/a/6TFwupl)

aryanpingle committed:
Add focus effects to inputs

Provides better keyboard accessibility for:
* Checkboxes
* Range Inputs
* Copy-over Buttons
* Option-reveal Elements
Along with some minor design improvements

aryanpingle committed:
Make text-fields match the dark theme

aryanpingle commented:
Hi @argyleink, do the changes look good to you? Before/After comparisons are in the description, and I've made the branch up-to-date with `dev` :)

argyleink commented:
sorry for the delay here, I got super sick last week and was out for a few days. I'm ramping back up this week and have this on my todo list 👍🏻

aryanpingle commented:
@argyleink Ahh no worries at all, hope you're feeling better now 😄

argyleink commented on src/client/lazy-app/Compress/Options/style.css:
```diff
@@ -118,6 +128,11 @@
     fill: var(--header-text-color);
   }
+
+  &:focus {
```
can you also add this new style to the new buttons (merged last week) `.save-button, .import-button`? a little akward atm that one has it and the other doesnt.

<img width="125" alt="Screenshot 2023-04-11 at 12 23 12 PM" src="https://user-images.githubusercontent.com/1134620/231267583-60ce78c7-1bfa-4afe-abe5-b83cdf761d0f.png">

aryanpingle replied:
Ahh nice catch, didn't see the new icons. I'll apply the same style to them.

argyleink reviewed (changes requested):
this is a significant improvement to the keyboard experience, thanks so  much! 

I've got 2 small requests 🙂 
1. I left one request inline with some code, there's 2 new buttons that could use the same focus treatment.
2. Could you also add a focus state to the close button? maybe it just goes opaque? or an outline works great too.

aryanpingle committed:
Add styles to save-import buttons

aryanpingle committed:
Add focus animation to back button

aryanpingle commented:
@argyleink I added some new changes, would love to hear your thoughts :)

1. I added the focus outline to the save/import buttons, and made them conditionally disabled (instead of just reducing the opacity)
2. The back button now has a simple blue pulse outline animation on focus. Nothing too fancy, just something to make it stand out against the light/dark backgrounds (image below)

![127 0 0 1_5500](https://user-images.githubusercontent.com/49340972/231401819-f9986121-fd36-4faf-84f2-57d98803b162.png)

argyleink reviewed (approved):
looks great!
so much better, ty for this 🙏🏻
